controllers/hetznercluster_controller_test.go

```diff
@@ -840,3 +842,100 @@ var _ = Describe("reconcileRateLimit", func() {
 		Expect(reconcileRateLimit(hetznerCluster, testEnv.RateLimitWaitTime)).To(BeFalse())
 	})
 })
+
+func TestSetControlPlaneEndpoint(t *testing.T) {
```

Comment:
leave some space between the different lines.. You have not a single empty line here. Leave one between every test, after you define the hetznerCluster, after you start the if statements with the tests, etc.